### PR TITLE
Improving the speed of some feature calculators

### PR DESCRIPTION
**Pull request by tdeboissiere**
The speed of some feature calculators can be improved with a different implementation (cf issue #148)

### CODE REVIEW

File: tsfresh/feature_extraction/feature_calculators.py [hunk 1]
```diff
@@ -614,8 +618,11 @@ def percentage_of_reoccurring_datapoints_to_all_datapoints(x):
     :return: the value of this feature
     :return type: float
     """
-    x = pd.Series(x)
-    return (x.value_counts() > 1).mean()
+
+    unique, counts = np.unique(x, return_counts=True)
+    counts[counts < 2] = 0
```
Comment (MaxBenChrist) [inline tsfresh/feature_extraction/feature_calculators.py]: I think one could change

```
counts[counts < 2] = 0
counts[counts > 0] = 1
np.sum(counts) / float(counts.shape[0])
```

to

```
np.sum(counts > 1)/float(counts.shape[0])
```

I guess it is more reader friendly.

File: tsfresh/feature_extraction/feature_calculators.py [hunk 2]
```diff
@@ -656,10 +663,10 @@ def sum_of_reoccurring_values(x):
...
     :return type: float
     """
-    x = pd.Series(x)
-    value_counts = x.value_counts()
-    doubled_values = value_counts[value_counts > 1]
-    return sum(doubled_values.index * doubled_values)
+
+    unique, counts = np.unique(x, return_counts=True)
+    counts[counts < 2] = 0
+    return np.sum(counts * unique)
```
Comment (MaxBenChrist) [inline tsfresh/feature_extraction/feature_calculators.py]: like above I would suggest to `np.sum(counts > 1)`

And the product with `unique` results in another feature actually.

**Comment by tdeboissiere** [comment]
Updated !

File: tsfresh/feature_extraction/feature_calculators.py [hunk 3]
```diff
@@ -656,10 +661,10 @@ def sum_of_reoccurring_values(x):
     :return: the value of this feature
     :return type: float
     """
-    x = pd.Series(x)
-    value_counts = x.value_counts()
-    doubled_values = value_counts[value_counts > 1]
-    return sum(doubled_values.index * doubled_values)
+
+    unique, counts = np.unique(x, return_counts=True)
```
Comment (MaxBenChrist) [inline tsfresh/feature_extraction/feature_calculators.py]: Sorry my last remark was wrong. This feature really expresses the sum of the recurring values.   

Can your reverse parts of last commit?  The following lines are fine for the `sum_of_reoccurring_values` feature

```
unique, counts = np.unique(x, return_counts=True)
counts[counts < 2] = 0
return np.sum(counts * unique)
```

**Comment by MaxBenChrist** [comment]
Thanks @tdeboissiere

**Pull request merged**